hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/DefaultAMSProcessor.java

```diff
@@ -334,6 +335,15 @@ public void allocate(ApplicationAttemptId appAttemptId,
         .pullJustFinishedContainers());
     response.setAvailableResources(allocation.getResourceLimit());
 
+    int totalVirtualCores = this.rmContext.getScheduler().getRootQueueMetrics()
```

Comment:
Extract this.rmContext.getScheduler().getRootQueueMetrics()

Reply:
Done and added null check also